PS8lite.Rmd

```diff
@@ -180,30 +175,84 @@ MiscVal + MoSold + YrSold" %>%
 
 ***
 
+# Fitting a `randomForest` via `caret`
+
+
+## Setup cross-validation
 
+We're going use `caret` package functionality to
 
-# Fit a Random Forest model
+1. Define cross-validation settings. We are going to use cross-validation to generate model error estimates (when using fitted models to make predictions on new independent test data)
+1. Setup a "search grid" of "tuning" parameters to find the "optimal value." By "optimize" we mean pick the value of the "tuning" parameter that yields the lowest cross-validated error estimate.
 
-Fit one Random Forest model `model_rf` using any approach you want.
+The latter is like how in the:
+
+1. CART problem set you searched over a range of $\alpha$ values
+1. LASSO problem set you searched over a range of $\lambda$ values
+
+The "tuning" parameter we're going to "optimize" over is the number of randomly chosen variables we use at each split in our CART trees: `mtry`. We're going to search over values 2 through 5 (the maximum number of predictor variables we have access to). 
 
 ```{r}
+# Define cross-validation settings: 10-fold CV
+fit_control <- trainControl(method = "cv", number = 10)
 
+# Setup search grid of "tuning" parameters
+mtry <- 2:10
+tunegrid <- expand.grid(.mtry = mtry)
 ```
 
 
+## Perform cross-validation
+
+Note that this code chunk takes a few minutes to run. So we set this code chucks `cache = TRUE` to save the result when knitting the .Rmd file, so that future knits don't re-run this code block to save time. 
```

Comment:
Indeed! This is a great example of a problem where computational constraints become an issue! In other words, method A might yield "better predictions" than method B, but if method A takes a week to run and method B takes only a few minutes, you might sacrifice a little predictive power for speed!